compiler/circle-mpqsolver/src/bisection/VISQErrorApproximator.cpp

```diff
@@ -17,23 +17,45 @@
 #include "VISQErrorApproximator.h"
 
 #include <fstream>
+#include <json.h>
 
 using namespace mpqsolver::bisection;
 
 void VISQErrorApproximator::init(const std::string &visq_data_path)
 {
   // read file
   std::ifstream file(visq_data_path);
-  if (!init(file))
+  try
+  {
+    init(file);
+  }
+  catch (std::runtime_error &)
   {
     throw std::runtime_error("Invalid visq file " + visq_data_path);
```

Comment:
If you're going to change `init(std::istream)` to throw exceptions (for better error messaging), it would be better to remove the try-catch statement.

Reply:
ok